nautobot_circuit_maintenance/__init__.py

```diff
@@ -49,8 +49,9 @@ def import_notification_sources(sender, **kwargs):  # pylint: disable=unused-arg
             slug=slugify(
                 notification_source["name"],
             ),
-            attach_all_providers=notification_source.get("attach_all_providers", False),
         )
+        instance.attach_all_providers = notification_source.get("attach_all_providers", False)
+        instance.save()
```

Comment:
Perhaps use `defaults` on `get_or_create` rather than a separate update and save step?

```python
        instance, _ = NotificationSource.objects.get_or_create(
            name=notification_source["name"],
            slug=slugify(
                notification_source["name"],
            ),
            defaults={
                "attach_all_providers": notification_source.get("attach_all_providers", False),
            }
        )
```

Reply:
I've double checked and using `defaults` is not updating well the desired stated of `attach_all_providers` when it was already set.

Reply:
Ah that makes sense - need to change this to use `update_or_create` instead of `get_or_create`.

Reply:
totally! (I was not aware of the `defaults` option, thanks)